addons/mail/static/src/js/composer.js

```diff
@@ -469,7 +469,16 @@ var BasicComposer = Widget.extend(chat_mixin, {
     preprocess_message: function () {
         // Return a deferred as this function is extended with asynchronous
         // behavior for the chatter composer
-        var value = _.escape(this.$input.val()).replace(/\n|\r/g, '<br/>');
+
+        //Remove Extra newlines from starting and ending of Message.
+        var value = _.escape(this.$input.val()).replace(/^(\r|\n)+|(\n|\r)+$/g, '');
+
+        //Combine 2 or more consecutive newlines into 1 newline.
+        value = value.replace(/(\r|\n){2,}/g, '<br/><br/>');
+
+        //convert newline to <br/> tag.
+        value = value.replace(/(\r|\n){1,}/g, '<br/>');
```

Comment:
Hello, Please refer this  
[Specification](https://docs.google.com/document/d/1zaJUfDBM7q4EnTqTGjxo4f5HtXtTHAQeu3AbqabejNA/edit)
2+ Line above are using to achieve 3rd Case in specs. While last line of 1+ uses to convert remaining '\n'. If i remove this line , Consecutive two line message will converted into single line , which is not desire output.